lib/topic_creator.rb

```diff
@@ -213,4 +239,29 @@ def add_groups(topic, groups)
   def check_can_send_permission!(topic, obj)
     rollback_with!(topic, :cant_send_pm) unless @opts[:skip_validations] || @guardian.can_send_private_message?(obj)
   end
+
+  def find_or_create_user(email, display_name)
+    user = nil
+
+    User.transaction do
```

Comment:
Is there a reason for the transaction here? I should be wrapped in a parent transaction anyway